docs/semgrep-ci/configuration-reference.md

```diff
@@ -22,20 +22,27 @@ SEMGREP_RULES="p/security-audit p/secrets"
 
 ## Diff-aware scanning (`SEMGREP_BASELINE_REF`)
 
-For [diff-aware scans](overview.md#features), set this variable
-to the git ref (branch name, tag, or commit hash) to use as a baseline.
-For example, to report findings newly added
-since branching off from your `main` branch, set
+For [diff-aware scans](overview.md#features), set this variable to create a baseline. This option limits scan results to those introduced after the git commit, in a branch, or tag. For example, you have a repository with 10 commits. You set the commit number 8 as the baseline. Consequently, Semgrep only returns scan results introduced by changes in commits 9 and 10.
 
-```sh
-SEMGREP_BASELINE_REF=main
-```
+NOTE: It is best to perform diff scans on branches other than your `main` branch. Performing a diff scan after a full-project scan of the same branch closes findings of the full-project. All findings of a particular branch are marked as fixed when the branch is re-scanned. As a consequence, Semgrep no longer reports their findings.
+
+### Examples of `SEMGREP_BASELINE_REF`
+
+To only report findings newly added
+since branching off from your `main` branch, set the following:
+    ```sh
+    SEMGREP_BASELINE_REF=$(git merge-base main HEAD)
+    ```
```

Comment:
I think we should add an info box explaining what this whole merge-base command is

Reply:
Here in the documentation or somewhere in our App? :-) Or do you have an idea of what to say in the infobox? If we could fix the underlying issue later on... we can avoid this. :D @underyx

Reply:
I'd explain here in the docs. I'd put the whiteboard drawing from earlier today into words somehow. I think we'd fix the underlying issue in 3-6 months maybe? Still worth documenting.